frontend/lib/identification/activation_workflow/activation_code_scanner_page.dart

```diff
@@ -100,4 +87,43 @@ class ActivationCodeScannerPage extends StatelessWidget {
       await showError("Ein unerwarteter Fehler ist aufgetreten.");
     }
   }
+
+  Future<void> _activateCode(
+    BuildContext context,
+    DynamicActivationCode activationCode,
+  ) async {
+    final client = GraphQLProvider.of(context).value;
+    final projectId = Configuration.of(context).projectId;
+    final provider = Provider.of<UserCodeModel>(context, listen: false);
+
+    final activationResult = await activateCode(
+      client: client,
+      projectId: projectId,
+      code: activationCode,
+      overwriteExisting: true,
+    );
+
+    switch (activationResult.activationState) {
+      case ActivationState.success:
+        if (activationResult.totpSecret != null) {
+          final totpSecret = const Base64Decoder().convert(activationResult.totpSecret!);
+          final userCode = DynamicUserCode(
+            info: activationCode.info,
+            pepper: activationCode.pepper,
+            totpSecret: totpSecret,
+          );
+          provider.setCode(userCode);
+        }
+        break;
+      case ActivationState.failed:
+        await QrParsingErrorDialog.showErrorDialog(
+          context,
+          "Der eingescannte Code ist ungültig.",
+        );
+        break;
+      case ActivationState.didNotOverwriteExisting:
+      // TODO: ask user to overwrite code
+      default:
```

Comment:
I think we should show an error here to the user in these two cases.

Reply:
not sure why `didNotOverwriteExisting` should show an exception.

Reply:
As it is not the expected outcome (the expected outcome would be `success`).
Once we implement the "ping-pong" (by asking the user whether the code should be overwritten) we can adjust this behavior. Until then, we should show an error...